lib/math.ex

```diff
@@ -472,4 +472,65 @@ defmodule Math do
   """
   @spec atanh(x) :: float
   defdelegate atanh(x), to: :math
+
+  @doc """
+  Computes the modular multiplicatibe inverse of `a` under modulo `m`
+
+  In other words, given integers `a` and `m` calculate a value `b` for `ab = 1 (mod m)`
+
+  Returns an `{:ok, b}` tuple or `{:error, :not_coprime}` when `b` cannot be calculated for the inputs.
+
+  ## Examples
+
+      iex> Math.mod_inv 3, 11
+      {:ok, 4}
+      iex> Math.mod_inv 10, 17
+      {:ok, 12}
+      iex> Math.mod_inv 123, 455
+      {:ok, 37}
+      iex> Math.mod_inv 123, 456
+      {:error, :not_coprime}
+
+  """
+  @spec mod_inv(a :: integer, m :: integer) :: {:ok, integer} | {:error, :not_coprime}
+  def mod_inv(a, m)
+
+  def mod_inv(a, m) do
+    1..(m - 1)
+    |> Enum.find(:not_coprime, &(rem(a * &1, m) == 1))
+    |> (fn
+          :not_coprime -> {:error, :not_coprime}
+          x -> {:ok, x}
+        end).()
+  end
+
+  @doc """
+  Computes the modular multiplicatibe inverse of `a` under modulo `m`
+
+  Similar to `mod_in/2`, but returns only the value or raises an error.
+
+  ## Examples
+
+      iex> Math.mod_inv! 3, 11
+      4
+      iex> Math.mod_inv! 10, 17
+      12
+      iex> Math.mod_inv! 123, 455
+      37
+      iex> Math.mod_inv! 123, 456
+      ** (ArithmeticError) Inputs are not coprime!
+
+  """
+  @spec mod_inv!(a :: integer, m :: integer) :: integer
+  def mod_inv!(a, m)
+
+  def mod_inv!(a, m) when is_float(a) or is_float(m),
```

Comment:
~~We should add a test case for this clause.~~ (You have one in the tests. That is indeed nicer since this is an exceptional input that we should not highlight in the documentation).

I am undecided whether we should add an `ArithmeticError` ("the expression does not make mathematical sense") for this or rather a plain `ArgumentError` ("the parameter(s) are not part of the domain of this function"), but leaning towards the latter.

Reply:
I also like `ArgumentError` better for this case. I believe it'll be more clear to the user that something is very wrong with the inputs as opposed to "these particular inputs don't express the mathematical property we'd like".